### PR TITLE
PyRuler Simon Game

### PR DESCRIPTION
**Pull request by isaacwellish**
reset branch, initial commit

### CODE REVIEW

File: PyRuler_Simon_Game/code.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+def record_caps(timeout=3):
+    start_time = time.monotonic() # start 3 second timer waiting for user input
+    while time.monotonic() - start_time < timeout:
+        caps = read_caps()
+        val = None
+        for i,c in enumerate(caps):
+            if c:
+                val = i
+                print("true!")
```
Comment (ladyada) [inline PyRuler_Simon_Game/code.py]: remove this print

File: PyRuler_Simon_Game/code.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                time.sleep(0.1)
+        if val is not None: # if there's input from a pad exit the timer loop
+            break
+    return val
+
+def light_cap(cap, duration=0.5):
+    # turn the LED for the selected cap on
+    leds[cap].value = True
+    # wait the requested amount of time
```
Comment (ladyada) [inline PyRuler_Simon_Game/code.py]: remove this comment, we know what time.sleep does

File: PyRuler_Simon_Game/code.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    time.sleep(duration)
+    # turn the LED for the selected region off
```
Comment (ladyada) [inline PyRuler_Simon_Game/code.py]: remove comment

File: PyRuler_Simon_Game/code.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    leds[cap].value = False
+    time.sleep(duration)
+
+def play_sequence(sequence):
+    duration = 1 - len(sequence) * 0.05
+    if duration < 0.1:
+        duration = 0.1
```
Comment (ladyada) [inline PyRuler_Simon_Game/code.py]: use math.max() here instead of a comparison

File: PyRuler_Simon_Game/code.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    for cap in sequence:
+        light_cap(cap, duration)
+
+def read_sequence(sequence):
+    pixels.fill(green)
+    pixels.show()
+    #time.sleep(1)
```
Comment (ladyada) [inline PyRuler_Simon_Game/code.py]: remove this if not used

File: PyRuler_Simon_Game/code.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    cap_touches[1] = touches[1].raw_value > 3000
+    cap_touches[2] = touches[2].raw_value > 3000
+    cap_touches[3] = touches[3].raw_value > 3000
+    return cap_touches
+
+def record_caps(timeout=3):
+    start_time = time.monotonic() # start 3 second timer waiting for user input
+    while time.monotonic() - start_time < timeout:
+        caps = read_caps()
+        val = None
```
Comment (ladyada) [inline PyRuler_Simon_Game/code.py]: this section is complex for what it does. check if cap is not None, and then return the first element if it isnt

File: PyRuler_Simon_Game/code.py (hunk 4 above)
Comment (isaacwellish) [reply PyRuler_Simon_Game/code.py]: How do you mean? max(list) returns the highest element in a list but don't know how I could use that here?

Comment (isaacwellish) [reply PyRuler_Simon_Game/code.py]: oh jk i figured it out!

Comment (caternuson) [reply PyRuler_Simon_Game/code.py]:
```python
duration = max(0.1, duration)
```

File: PyRuler_Simon_Game/code.py (hunk 6 above)
Comment (isaacwellish) [reply PyRuler_Simon_Game/code.py]: rearranged this a bit but not exactly sure if it's what you were thinking.

File: PyRuler_Simon_Game/code.py (hunk 4 above)
Comment (isaacwellish) [reply PyRuler_Simon_Game/code.py]: thx @caternuson!

**Comment by ladyada** [comment]
ok come by when you're done with your edits

**Comment by isaacwellish** [comment]
ok changes are ready just have one pylint issue I don't know how to resolve. When I changed the intro method to 

`for led in leds:
        led.value = True
        time.sleep(0.25)
    for led in leds:
        led.value = False`

pylint gives me the following error:

`Redefining name 'led' from outer scope (line 23) (redefined-outer-name)`

**Comment by isaacwellish** [comment]
Ok I just took out the intro function and put the code at the beginning of the main loop

**Comment by ladyada** [comment]
ok can merge once travis passes

**Comment by isaacwellish** [comment]
yay!

**Comment by isaacwellish** [comment]
can you merge when you get a chance please?

**Pull request merged**